bluetooth-rfkill-event/bluetooth_rfkill_event.c

```diff
@@ -913,17 +948,51 @@ void attach_hci()
 {
     char hci_execute[PATH_MAX * 2];
     int r;
+    FILE *fp_btwrite = NULL;
 
     DEBUG("");
 
     snprintf(hci_execute, sizeof(hci_execute), "%s %s", hciattach, hciattach_options);
 
-    r = system_timeout(hci_execute);
-    INFO("executing %s %s", hci_execute,
-         (WIFEXITED(r) && !WEXITSTATUS(r)) ? "succeeded" : "failed");
+    if (main_opts.bt_reset_btwrite) {
+        fp_btwrite = fopen("/proc/bluetooth/sleep/btwrite", "w");
+        if (!fp_btwrite) {
+            fprintf(stderr, "Failed to open /proc/bluetooth/sleep/btwrite\n");
+        } else {
+            fwrite("0\n", 3, 1, fp_btwrite);
+            fflush(fp_btwrite);
+            // sleep 500ms
+            usleep(500*1000);
+            fwrite("1\n", 3, 1, fp_btwrite);
+            fflush(fp_btwrite);
+            // sleep 500ms
+            usleep(500*1000);
+        }
+        fclose(fp_btwrite);
+    }
+
+    if (main_opts.no_wait) {
+        if (fork() == 0) {
+            char *argv[4] = {
+                "/bin/sh",
+                "-c",
+                hci_execute,
+                NULL
+            };
+            if (execv("/bin/sh", argv) < 0) {
+                WARN("Can't execute command '%s': %s(%d)",
+                 hci_execute, strerror(errno), errno);
+            }
+            exit(EXIT_FAILURE);
```

Comment:
The fork always returns EXIT_FAILURE, does this matter?

Reply:
If execve succeeds that exit will never be reached